app/models/apple/podcast_delivery.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+module Apple
+  class PodcastDelivery < ActiveRecord::Base
+    include Apple::ApiResponse
+
+    serialize :api_response, JSON
+
+    has_many :podcast_delivery_files
+    belongs_to :episode, class_name: '::Episode'
+    belongs_to :podcast_container, class_name: '::Apple::PodcastContainer'
+
+    delegate :apple_episode_id, to: :podcast_container
+
+    enum status: {
+      awaiting_upload: 'AWAITING_UPLOAD',
+      completed: 'COMPLETED',
+      failed: 'FAILED'
+    }
+
+    def self.update_podcast_deliveries_state(api, episodes)
+      podcast_containers = episodes.map do |ep|
+        if ep.podcast_container.present?
+          ep.podcast_container
+        else
+          Rails.logger.error("Missing podcast container for episode #{ep.feeder_id}")
+          nil
+        end
+      end.compact
+
+      results = get_podcast_deliveries_via_containers(api, podcast_containers)
```

Comment:
Most of the models follow this pattern for updating existing state:

1) Use the parent model to query for children
2) Reformat the results by querying the main api endpoint for the child model.